### ISSUE TITLE
I constantly reload the page goes

### ISSUE THREAD

**Issue by heiheshang** [zotonic/zotonic#1512]
Look at my problem. If I try to login user error occurs and all sessions are lost. I have to re-login administrator. there are no errors.A normal user can not do the login, only the administrator\
https://cloud.mail.ru/public/3auK/XpPdNAKkw

**Comment by mworrell** [comment]
@ddeboer can you have a look at this as well?

@heiheshang you are running the docker image with 0.22, aren't you?

**Comment by heiheshang** [comment]
I work in the docker, not using your. Release 0.22 is really

**Comment by ddeboer** [comment]
@heiheshang Do you see any messages in the console or the logs?

**Comment by heiheshang** [comment]
There are no posts
https://cloud.mail.ru/public/9kVU/ECsiJFCYC

**Comment by ddeboer** [comment]
Okay, so nothing in the logs. And in the JS console? 

Was your application on 0.21.0 before?

**Comment by heiheshang** [comment]
I did not do requests. The js console can be seen on video error

**Comment by ddeboer** [comment]
Are your sure it’s not just some browser cache issue? What happens if you try to log in in an incognito window or another browser?

**Comment by ddeboer** [comment]
What’s the last Zotonic version that your application ran successfully on?

In the [diff between 0.21.0 and 0.22.0](https://github.com/zotonic/zotonic/compare/0.21.0...0.22.0) I see only two candidates that *may* interfere with login: 12a35fb2a7a26488d1b3ac42fedab2daa6d33a50 and 3ce1b8afdda82f4e25fbde05dbc406bbdc1ccc14.

**Comment by heiheshang** [comment]
At 21, the same did not work, so I upgraded to 22

**Comment by heiheshang** [comment]
I started with 20

**Comment by ddeboer** [comment]
And on 0.20.0 everything worked normally?

**Comment by heiheshang** [comment]
yes

**Comment by heiheshang** [comment]
Authorization may conclude more informative error? I do not understand what Fatal: error

**Comment by ddeboer** [comment]
And you’re saying that the same error already occurred at version 0.21.0? I’m asking because this may help us pinpoint the code change that causes the error.

> I do not understand what Fatal: error

You mean the ‘FAIL: error’ in the top right of the browser window? Yeah, it doesn’t really help. :wink:

**Comment by heiheshang** [comment]
Yes, I tried to 0.21.1

**Comment by heiheshang** [comment]
Firefox не может установить соединение с сервером ws://amurasso.anriya.ru:8000/websocket.
http://amurasso.anriya.ru:8000/lib/js/apps/zotonic-1.0~z.widgetmanager~/js/modules/ubf~z.notice~z.imageviewer~z.dialog~z.clickable~livevalidation-1.3~z.inputoverlay~jquery.loadmask~476795580.js
Line 1198

**Comment by heiheshang** [comment]
a regular user does not have permission to create a socket. I think it is necessary to assign rights, but do not know what I have tried to give the user the right to all of the modules, but it did not help.

**Comment by ddeboer** [comment]
If you give the regular user `use` permission on `mod_admin`, can that user then login on the `/admin` login screen?

**Comment by heiheshang** [comment]
If you just give the user the right to mod_admin it does not help.

**Comment by ddeboer** [comment]
What error do you get? Users with `use mod_admin` permissions should be able to log in at `http://yoursite.com/admin`.

**Comment by heiheshang** [comment]
an ordinary user can not log in admin

**Comment by heiheshang** [comment]
the same mistake

**Comment by heiheshang** [comment]
and after an unsuccessful attempt to login, the administrator session dies, requires a new login

**Comment by ddeboer** [comment]
Could this be related to #1511?

**Comment by heiheshang** [comment]
No, my problem is different. I removed nginx and work through the 8000 port. I have no problem with the admin, I have a problem with the ordinary user. I have included a module mod_acl_user_groups

**Comment by heiheshang** [comment]
I installed the 0.20 release, and everything worked.

**Comment by ddeboer** [comment]
> I installed the 0.20 release, and everything worked.

Okay, in the [comparison between 0.20.0 and 0.21.0](https://github.com/zotonic/zotonic/compare/0.20.0...0.21.0) nothing jumps out that should cause this behaviour. @mworrell and @mmzeeman Can you have a look at your own changes? Can any of those cause this error?

**Comment by mmzeeman** [comment]
Could it be the bcrypt dep? Is that compiled?

We switched the password hash algorithm to bcrypt. When users login their hashes are upgraded to bcrypt automatically. I'm curious if the bcrypt dep was compiled correctly.

Could you check if the following files are available?

```
$ ls -l deps/bcrypt/priv/
total 376
-rwxr-xr-x  1 maas  staff   86948 Sep  5 08:54 bcrypt
-rwxr-xr-x  1 maas  staff  101148 Sep  5 08:54 bcrypt_nif.so
```

**Comment by heiheshang** [comment]
I now can not, it is necessary to switch branch

**Comment by mmzeeman** [comment]
It would explain why you could login as admin, but not as a regular user. The admin's password is verified in a different way than the user passwords. 

It could be that the nif was not compiled correctly for some unknown reason.

**Comment by mworrell** [comment]
@heiheshang is this still a problem with the 0.x (or master)?

**Comment by heiheshang** [comment]
this is a problem connecting javascript
Not all scripts have been connected


>Пятница, 13 января 2017, 0:14 +09:00 от Marc Worrell <notifications@github.com>:
>
>@heiheshang is this still a problem with the 0.x (or master)?
>—
>You are receiving this because you were mentioned.
>Reply to this email directly,  view it on GitHub , or  mute the thread .


-- 
microstoun microstoun

**Comment by mworrell** [comment]
Not so long ago I fixed an issue where a page could be repeatedly reloading if the page tries to subscribe to a topic before the page is initialized. Do you still see the same problem with the current 0.x?

**Comment by heiheshang** [comment]
no,but still I use the 0.22 release


>Пятница, 13 января 2017, 16:50 +09:00 от Marc Worrell <notifications@github.com>:
>
>Not so long ago I fixed an issue where a page could be repeatedly reloading if the page tries to subscribe to a topic before the page is initialized. Do you still see the same problem with the current 0.x?
>—
>You are receiving this because you were mentioned.
>Reply to this email directly,  view it on GitHub , or  mute the thread .


-- 
microstoun microstoun
